nextjs-task-system/app/api/auth/login/route.ts

```diff
@@ -0,0 +1,108 @@
+import { NextResponse } from "next/server";
+import db from "@/Utils/db";
+import { cookies } from "next/headers";
+import bcrypt from "bcryptjs";
+
+interface User {
+  id: number;
+  email: string;
+  first_name: string;
+  last_name: string;
+  role: string;
+  password: string;
+}
+
+export async function POST(request: Request) {
+  try {
+    const body = await request.json();
+    const { email, password } = body;
+
+    const user = db
+      .prepare(
+        "SELECT id, email, first_name, last_name, role, password FROM users WHERE email = ?",
+      )
+      .get(email) as User | undefined;
+
+    if (!user) {
+      return NextResponse.json(
+        { error: "Invalid credentials" },
+        { status: 401 },
+      );
+    }
+
+    const isValidPassword = await bcrypt.compare(password, user.password);
+    if (!isValidPassword) {
+      return NextResponse.json(
+        { error: "Invalid credentials" },
+        { status: 401 },
+      );
+    }
+
+    const sessionId = crypto.randomUUID();
+    db.prepare(
+      "INSERT INTO sessions (id, user_id, created_at) VALUES (?, ?, datetime('now'))",
+    ).run(sessionId, user.id);
+
+    const response = NextResponse.json({
+      id: user.id,
+      email: user.email,
+      firstName: user.first_name,
+      lastName: user.last_name,
+      role: user.role,
+      redirectTo: user.role === "admin" ? "/admin" : "/user/tasks",
+    });
+
+    response.cookies.set("sessionId", sessionId, {
+      httpOnly: true,
+      secure: process.env.NODE_ENV === "production",
+      sameSite: "lax",
+      path: "/",
+    });
```

Comment:
👍 Excellent cookie security settings:

```
response.cookies.set("sessionId", sessionId, {
      httpOnly: true,                                    // Prevents JavaScript access
      secure: process.env.NODE_ENV === "production",     // Forces HTTPS in production
      sameSite: "lax",                                  // Protects against CSRF
      path: "/",                                        // Proper scope setting
    });
```

Reply:
Potential improvement: Consider adding maxAge or expires option
to match your 24-hour session expiration logic in the GET route